nhs/data/filter.py

```diff
@@ -1,26 +1,147 @@
-from typing import List
+import glob
 
 import polars as pl
 
 
+def load_gnaf_files_by_states(
+    gnaf_path: str, states: list[str] = []
+) -> tuple[pl.LazyFrame, pl.LazyFrame]:
+    """
+    Load and filter ADDRESS_DEFAULT_GEOCODE and ADDRESS_DETAIL files for the specified states
+    from the given GNAF directory, and return them as LazyFrames.
+
+    Parameters:
+    - gnaf_path: str, the directory path where the GNAF psv files are stored.
+    - states: list[str], an optional list of state/territory names (e.g., ["WA", "ACT"]).
+              If an empty list is provided, the function will default to including all states.
+
+    Returns:
+    - tuple: (default_geocode_lf, address_detail_lf)
+      - default_geocode_lf: pl.LazyFrame, the merged ADDRESS_DEFAULT_GEOCODE data for the specified states,
+                            with an added "STATE" column.
+      - address_detail_lf: pl.LazyFrame, the merged ADDRESS_DETAIL data for the specified states.
```

Comment:
change to pandas style docstring, see other functions for example

Reply:
updated